### PR TITLE
Only enable the failover fix if configured

### PR DESCRIPTION
**Pull request by packi**
To be able to roll it out in a more controlled fashion

[BF-1847]

[BF-1847]: https://aiven.atlassian.net/browse/BF-1847?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: pglookout/pglookout.py [hunk 1]
```diff
@@ -483,29 +485,38 @@ def consider_failover(self, own_state, master_node, standby_nodes):
...
-                if master_known_to_be_gone or config_timeout_exceeded:
-                    # we've seen a master at some point in time, but now it's
-                    # not reachable or removed from configuration, perform an
-                    # immediate failover to promote one of the standbys
-                    if master_known_to_be_gone:
-                        reason = "master got removed from configuration"
-                    else:
-                        reason = "master node is not reachable"
-                    self.log.warning("Performing failover decision because %s", reason)
+            if master_node and not self._failover_on_disconnect:
```
Comment (Mulugruntz) [inline pglookout/pglookout.py:488]:
```suggestion
            if not self._failover_on_disconnect and master_node:
```

To only check iff the feature is enabled.

**Review by Mulugruntz — COMMENTED** [review]
We have a test to test the new feature flag mechanism, but we don't have a test for the new feature itself? (`if master_node`)
To make sure that it behaves differently than previous version.

**Comment by packi** [comment]
The test I've adjusted should cover the feature and the disabling of it.

File: pglookout/pglookout.py [hunk 2]
```diff
@@ -483,7 +485,10 @@ def consider_failover(self, own_state, master_node, standby_nodes):
                 self.current_master,
                 master_node,
             )
-            if self.current_master:
+            if not self._failover_on_disconnect and master_node:
+                self.stats.increase("failover_decision_on_disconnect_not_taken")
+                self.log.warning("Not considering failover, because it's not enabled by configuration")
+            elif self.current_master:
```
Comment (Mulugruntz) [inline pglookout/pglookout.py:491]: About the last line of this method:
```python
self.check_replication_lag(own_state, standby_nodes)
```

This is now reached in either of the following conditions:
- `not self._failover_on_disconnect and master_node`
- `self.current_master and not master_known_to_be_gone and not config_timeout_exceeded`

Is that what we want? Or do we want to skip this check in our new case (master and flag disabled)? If so, we might want to do this instead:

```suggestion
                return
            if self.current_master:
```

**Review by Mulugruntz — APPROVED** [review]

**Pull request merged**